ci/kokoro/docker/build-in-docker-bazel.sh

```diff
@@ -97,44 +97,119 @@ if [[ "${RUN_INTEGRATION_TESTS}" == "yes" || \
 
   # Changing the PATH disables the Bazel cache, so use an absolute path.
   readonly GCLOUD="/usr/local/google-cloud-sdk/bin/gcloud"
+  readonly GCLOUD_CONFIG="cloud-cpp-integration"
+
+  cleanup_hmac_service_account() {
+    local -r ACCOUNT="$1"
+    log_normal "Deleting account ${ACCOUNT}"
+    # We can ignore errors here, sometime the account exists, but the bindings
+    # are gone (or were never created). The binding is harmless if the account
+    # is deleted.
+    "${GCLOUD}" --quiet projects remove-iam-policy-binding \
+        "${GOOGLE_CLOUD_PROJECT}" \
+        --member "serviceAccount:${ACCOUNT}" \
+        --role roles/iam.serviceAccountTokenCreator >/dev/null || true
+    "${GCLOUD}" --quiet iam service-accounts delete "${ACCOUNT}" >/dev/null
+  }
+
+  cleanup_stale_hmac_service_accounts() {
+    # The service accounts created below start with hmac-YYYYMMDD-, we list the
+    # accounts with that prefix, and with a date from at least 2 days ago to
+    # find and remove any stale accounts.
+    local THRESHOLD="$(date +%Y%m%d --date='2 days ago')"
+    readonly THRESHOLD
+    local email
+    "${GCLOUD}" --quiet iam service-accounts list \
+        --project="${GOOGLE_CLOUD_PROJECT}" \
+        --filter="email~^hmac-[0-9]{8}- AND email<hmac-${THRESHOLD}-" \
+        --format='csv(email)[no-heading]' | \
+    while read -r email; do
+      cleanup_hmac_service_account "${email}"
+    done
+  }
+
+  create_hmac_service_account() {
+    local -r ACCOUNT="$1"
+    local -r EMAIL="${ACCOUNT}@${GOOGLE_CLOUD_PROJECT}.iam.gserviceaccount.com"
+    "${GCLOUD}" --quiet iam service-accounts create \
+        "--project=${GOOGLE_CLOUD_PROJECT}" "${ACCOUNT}"
+    log_normal "Grant service account permissions to create HMAC keys."
+    "${GCLOUD}" --quiet projects add-iam-policy-binding \
+        "${GOOGLE_CLOUD_PROJECT}" \
+        --member "serviceAccount:${EMAIL}" \
+        --role roles/iam.serviceAccountTokenCreator >/dev/null
+  }
+
+  trap restore_default_gcloud_config EXIT
+  restore_default_gcloud_config() {
+    "${GCLOUD}" --quiet config configurations activate default
+    "${GCLOUD}" --quiet config configurations delete "${GCLOUD_CONFIG}"
+  }
 
   echo
   echo "================================================================"
-  log_normal "Create an access token to run the Bigtable credential examples."
+  if "${GCLOUD}" --quiet config configurations \
+           describe "${GCLOUD_CONFIG}" >/dev/null 2>&1; then
+    log_normal "Activate the gcloud configuration for the cloud-cpp tests."
+    "${GCLOUD}" --quiet --no-user-output-enabled config configurations \
+        activate "${GCLOUD_CONFIG}"
```

Comment:
Can we use `--configuration=${GCLOUD_CONFIG}` on subsequent commands, or `export CLOUDSDK_ACTIVE_CONFIG_NAME=${GCLOUD_CONFIG}` to affect them all,  in order to avoid having to change global state via `activate` (and then having to guess that `activate default` will reverse that)?

Reply:
Great idea... let me take a stab at it later tonight or tomorrow (in this PR, of course).

Reply:
Done.